src/tests/rpp/test_flat_map.cpp

```diff
@@ -105,22 +106,43 @@ TEMPLATE_TEST_CASE("flat_map", "", rpp::memory_model::use_stack, rpp::memory_mod
                 CHECK(mock.get_on_error_count() == 1);
             }
         }
+        SECTION("subscribe using flat_map with templated lambda")
+        {
+            obs | rpp::operators::flat_map([](auto v) { return rpp::source::just(v * 2); })
+                | rpp::ops::subscribe(mock.get_observer());
+            SECTION("observer obtains values from underlying observables")
+            {
+                CHECK(mock.get_on_completed_count() == 1);
+            }
+        }
     }
-    /*
+}
+
+TEST_CASE("flat_map copies/moves")
+{
     SECTION("flat_map doesn't produce extra copies")
     {
         copy_count_tracker verifier{};
-        auto obs = rpp::source::just<TestType>(std::move(verifier))
-                 | rpp::ops::flat_map([](copy_count_tracker&& verifier) { return rpp::source::just<TestType>(std::move(verifier)); });
-        SECTION("subscribe")
+        auto obs = rpp::source::create<copy_count_tracker>([verifier = std::move(verifier)](const auto& obs) { obs.on_next(verifier); })
+                 | rpp::ops::map([](copy_count_tracker verifier) { return std::move(verifier); }) // copy from source to map
```

Comment:
try instead
```cpp
 auto obs = rpp::source::create<copy_count_tracker>([verifier = std::move(verifier)](const auto& obs) { obs.on_next(std::move(verifier)); })
```

Reply:
Though I subscribe two times here, so verifier will be in an invalid state in second subscription as it was already moved no?

Reply:
I changed to capture by ref and pass by value

Reply:
> Though I subscribe two times here, so verifier will be in an invalid state in second subscription as it was already moved no?

No, verifier is special class that is valid after move =)